includes/class-alg-wc-pif-core.php

```diff
@@ -136,17 +136,29 @@ public function delete_order_file_uploads( $postid ) {
 		 */
 		public function handle_downloads() {
 			if ( isset( $_GET['alg_wc_pif_download_file'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
-				$file_name  = $_GET['alg_wc_pif_download_file']; // phpcs:ignore
+				$allowd_types = array( 'jpg', 'jpeg', 'gif', 'png', 'mp3', 'mp4', 'doc', 'docx', 'xlsx', 'xls', 'ppt', 'pptx' );
+
+				// Filter alg_wc_pif_download_extra_extensions to add extra extensions.
+				$extra_ext  = apply_filters( 'alg_wc_pif_download_extra_extensions', $extra_ext, $arg );
+				$extensions = array_merge( $allowd_types, $extra_ext );
+				$file_name  = sanitize_text_field( wp_unslash( $_GET['alg_wc_pif_download_file'] ) ); // phpcs:ignore
+				$file_type  = wp_check_filetype( $file_name );
```

Comment:
we will not need this part since we already have restrictions on the frontend when uploading the file. Here are the related settings: https://prnt.sc/tqezng

Reply:
@dhruvins The attached image you are referring to is for "Upload" restriction which are in-place as per our discussion. But the security vulnerability issue is related to file download which Admin can only download.

And the code you are talking about is for Download not for the upload, also it's not related to the global settings input file field too.

Could you please clarify on this issue.

Reply:
agreed, we can keep this code for additional protection when the file is getting downloaded. But the additional condition about the file type present or not is not allowing me to download the file at all. Additionally, determining the product fields and extensions allowed at Product level will be difficult to get.